Database read and write performed

Swasti008 opened the pull request:
**Steps taken:-** 
1. Server **connection to MongoDB** using mongoose library.
2. **Read route performed** to get the profile picture for specific userID.
3. **Write route performed** to patch the profile picture for specific userID.

Swasti008 committed:
Database read and write performed

Swasti008 commented:
@CodiumAI-Agent /review

QodoAI-Agent commented:
## PR Review 🔍

<table>
<tr>
<tr><td> ⏱️&nbsp;<strong>Estimated&nbsp;effort&nbsp;to&nbsp;review [1-5]</strong></td><td>

3, because the PR involves changes to database connection and handling routes which requires careful review of the database operations and error handling mechanisms. The changes are not extensive but are critical due to their impact on data integrity and application stability.


</td></tr>
<tr><td> 🧪&nbsp;<strong>Relevant tests</strong></td><td>

No


</td></tr>
<tr><td> ⚡&nbsp;<strong>Possible issues</strong></td><td>

<strong>Possible Bug:</strong> The new routes do not handle the case where a user with the specified `userId` does not exist. This could lead to unhandled exceptions or incorrect API responses.

</td></tr>
<tr><td> 🔒&nbsp;<strong>Security concerns</strong></td><td>

No

</td></tr>
</table>


<details><summary> <strong>Code feedback:</strong></summary>

<hr><table><tr><td>relevant file</td><td>server/db.js
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Consider using a more secure method to handle MongoDB connection failures. Instead of just logging the error, you might want to implement a retry mechanism or alert the system administrators. This is important to ensure the database connectivity is robust and reliable. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S59_Swasti_Capstone_OrbitIQ/pull/29/files#diff-faec190164d3315987d61df45930a6c96e6ad07a9a2b8e0a63e7e79d0d91153cR7'>await mongoose.connect(process.env.URI, {</a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>server/src/Routes/fileUpload.js
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Add error handling for the case when `User.findById(userId)` does not find a user. This could prevent sending a 500 error when a 404 (Not Found) might be more appropriate. [important]

</strong>
</td></tr><tr><td>relevant line</td><td>const user = await User.findById(userId);</td></tr></table><hr>

<table><tr><td>relevant file</td><td>server/src/Routes/fileUpload.js
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Validate the `userId` parameter in the route to ensure it is a valid MongoDB ObjectId before attempting to find the user. This can prevent potential crashes or unexpected behavior when invalid IDs are provided. [medium]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S59_Swasti_Capstone_OrbitIQ/pull/29/files#diff-b330a770e4f09b0190a5f2f64fd21a2fe44e7c893f264bad7110118b6d098387R25'>const userId = req.params.userId;</a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>server/src/Routes/fileUpload.js
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Implement file type and size validation in the multer configuration to prevent uploading of potentially dangerous files. This is crucial for maintaining the security and integrity of the server. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S59_Swasti_Capstone_OrbitIQ/pull/29/files#diff-b330a770e4f09b0190a5f2f64fd21a2fe44e7c893f264bad7110118b6d098387R20'>const upload = multer({ storage: storage, validateFile: validateFile });</a></td></tr></table><hr>

</details>

Swasti008 committed:
Action performed on AI-review

Swasti008 commented:
Action taken on AI-review:- 
1. added retry functionality in connecting the Server to MongoDB.
2. checked the userID is valid MongoDB ObjectId or not 
3. added size validation in the multer
4. added error handling for the case when User.findById(userId) is not found.
